File: MCTabBarDemo/MCTabBarController/Example/ViewController.m
//
//  ViewController.m
//  MCTabBarController
//
//  Created by caohouhong on 2018/12/7.
//  Copyright © 2018年 caohouhong. All rights reserved.
//

#import "ViewController.h"
#import "PushViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    UIColor *randomColor = [UIColor colorWithRed:(arc4random()%255)*1.0f/255.0
                                           green:(arc4random()%255)*1.0f/255.0
                                            blue:(arc4random()%255)*1.0f/255.0 alpha:1];
    self.view.backgroundColor = randomColor;
    
    UIButton *btn = [[UIButton alloc] initWithFrame:CGRectMake(0, 0, 150, 80)];
    btn.center = self.view.center;
    [btn setTitle:@"按钮" forState:UIControlStateNormal];
    [btn setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
    [btn setBackgroundColor:[UIColor orangeColor]];
    btn.layer.cornerRadius = 4.0;
    [btn addTarget:self action:@selector(btnAction) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:btn];
    
    UILabel *label = [[UILabel alloc] initWithFrame:CGRectMake(0, 20, self.view.frame.size.width, 20)];
    label.text = @"点击空白区域退出";
    label.textColor = [UIColor whiteColor];
    label.font = [UIFont systemFontOfSize:13];
    label.textAlignment = NSTextAlignmentCenter;
    [self.view addSubview:label];
}

- (void)btnAction{
    
    [self.navigationController pushViewController:[[PushViewController alloc] init] animated:YES];
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event {
    [self.navigationController dismissViewControllerAnimated:YES completion:nil];
}

@end

File: MCTabBarControllerSwift/ViewController.swift
//
//  ViewController.swift
//  MCTabBarControllerSwift
//
//  Created by caohouhong on 2018/12/11.
//  Copyright © 2018年 chh. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        let randomColor = UIColor(red: CGFloat((arc4random()%255))/255.0, green: CGFloat((arc4random()%255))/255.0, blue: CGFloat((arc4random()%255))/255.0, alpha: 1)
        self.view.backgroundColor = randomColor
        
        let button = UIButton(frame: CGRect(x: 0, y: 0, width: 150, height: 80))
        button.center = self.view.center
        button.layer.cornerRadius = 5.0
        button.setTitle("按钮", for: .normal)
        button.setTitleColor(UIColor.white, for: .normal)
        button.backgroundColor = UIColor.orange
        button.addTarget(self, action: #selector(buttonAction), for: .touchUpInside)
        self.view.addSubview(button)
        
        let label = UILabel(frame: CGRect(x: 0, y: 10, width: self.view.frame.size.width, height: 20))
        label.textColor = UIColor.white
        label.text = "点击空白区域退出"
        label.font = UIFont.systemFont(ofSize: 12)
        label.textAlignment = .center
        self.view.addSubview(label)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

    @objc func buttonAction(){
        self.navigationController?.pushViewController(PushViewController(), animated: true)
    }

    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        self.navigationController?.dismiss(animated: true, completion: nil)
    }
}

